src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1015,9 +1016,12 @@ protected virtual void OnDocumentRemoved(string filePath)
         {
         }
 
-        protected virtual void UpdateAnalyzerRules()
-        {
-        }
+        /// <summary>
+        /// Implemented by derived types to provide a way for <see cref="AbstractProject"/> to indicate that options will need to be refreshed.
+        /// It is expected that derived types will read in shared option state stored in this class, create new Compilation and Parse options,
+        /// and call <see cref="SetOptions"/> in response.
+        /// </summary>
+        protected abstract void UpdateOptions();
```

Comment:
It looks like the TypeScriptProject class is not giving us anything but a way to capture some configuration options... the TypeScriptProject only has a constructor, we don't override/implement any methods.

Is it an option to make the abstractproject class, non-abstract?